src/app/patient-dashboard/formentry/previous-encounter-resolver.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable }             from '@angular/core';
+import {
+  Router, Resolve, RouterStateSnapshot,
+  ActivatedRouteSnapshot
+} from '@angular/router';
+
+import { PatientPreviousEncounterService } from '../patient-previous-encounter.service';
+import { FormSchemaService } from './form-schema.service';
+
+@Injectable()
+export class PreviousEncounterResolver implements Resolve<any> {
+
+  constructor(private patientPreviousEncounterService: PatientPreviousEncounterService,
+              private formSchemaService: FormSchemaService) {
+  }
+
+  resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<any> | any {
+
+    let selectedFormUuid = route.params['formUuid'];
+    console.log('----FormUuid', selectedFormUuid);
+
+    return new Promise((resolve, reject) => {
+
+      this.formSchemaService.getFormSchemaByUuid(selectedFormUuid).subscribe(
+        (compiledFormSchema) => {
+          console.log('compiledFormSchema', compiledFormSchema);
+          if (compiledFormSchema) {
```

Comment:
Usually, when viewing an existing encounter, one does not need to view 'historical values', and therefore fetching it delays loading of a saved encounter. Add a check to see of there's an encounter uuid in the url, and if present, skip the loading previous encounter part